### ISSUE TITLE
Replace the existing US-located numbers, date and amounts with locale-aware components/ methods 

### ISSUE THREAD

**Issue by jboniface** [Expensify/Expensify.cash#2349]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/master/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
We are implementing the structure for internationalisation and localisation in Expensify.Cash.

As part of that, we need to replace the existing US-located numbers, dates, amounts and phone numbers with locale-aware components/ methods. In order to do that, you need to identify all places that display any of that information and then use the methods `numberFormat`, `timestampToRelative`, `timestampToDateTime`, `toLocalPhone`, `fromLocalPhone` of the `withLocalize` HOC we added [here](https://github.com/Expensify/Expensify.cash/pull/2208).

## Expected Result:
numbers, dates, and amounts should be locale-aware

## Actual Result:
They are US centric

## Platform:
All

**Version Number:** current
https://github.com/Expensify/Expensify/issues/152154

## Job on Upwork
view Upwork job [here](https://www.upwork.com/jobs/~01aa2b8780a38644a6)

**Comment by Andmat7** [comment]
Good day, my proposed solution:
1. I would start updating the dates that are easier to test:
  a) I would search file by file looking for the dates and understanding how they work, 
   b) then replace them with the HOA or do the necessary small refactorings. 
2. the same procedure as above with the dates and finally with the amounts.
Obviously, I would test by test and at the end, I would verify that it compiles and that everything works correctly.

I could send pull requests for each type for review: a pull request for dates, another pull request for numbers, etc, or all at once, as you tell me.

**Comment by iwiznia** [comment]
> b) then replace them with the HOA or do the necessary small refactorings.

Sorry, what's HOA?

Also, can you provide a list of places you'll update?

**Comment by iwiznia** [comment]
Yeah, you can link to every text/file that will need updating

**Comment by Andmat7** [comment]
Well, I have to search file by file, and it's part of the job, do you consider this list essential to start the upwork contract? ,
Do you find better just a couple of examples and after starting the contract the complete list?

**Comment by iwiznia** [comment]
Mainly I was asking for the list to make sure we are not missing any place to update and that you are not on the hook for any new place we use numbers/dates/amounts that get added after your PR is over.
If you don't want to create the list now, that's fine, but we will be asking for followup PRs for any places that are missing, before paying. Let me know how you want to proceed.

**Comment by Andmat7** [comment]
Sorry for the inconvenience, please start the contract, with that I feel calm and I focus on making those changes as fast as possible, at the end, you must review and approve that everything is ok before closing the contract. 

If you want my first task is to list the files and their respective lines before starting to write the code. 

For example in this part
https://github.com/Expensify/Expensify.cash/blob/9fbf82e8237b7d1ff416619970038cc0d5bf9252/src/pages/iou/steps/IOUAmountPage.js#L71

I am going to use NumberFormat to enable language sensitive number formatting

**Comment by iwiznia** [comment]
That line is not rendering the amount, so that would not be the place to localize it, it would be [here](https://github.com/Expensify/Expensify.cash/blob/9fbf82e8237b7d1ff416619970038cc0d5bf9252/src/pages/iou/steps/IOUAmountPage.js#L107)

**Comment by Andmat7** [comment]
ok only rendering,

**Comment by iwiznia** [comment]
@Andmat7 I see you deleted several of your comments, any reason why? If you want to work on this, please re-post here your proposal.

**Comment by parasharrajat** [comment]
Here is an outline
1. We will use `withLocalize` HOC to use the numberFormat, timestampToRelative, timestampToDateTime, toLocalPhone, fromLocalPhone method.
2. We replace all the places where we see number and time date with the following method calls 
e.g.
https://github.com/Expensify/Expensify.cash/blob/9fbf82e8237b7d1ff416619970038cc0d5bf9252/src/pages/iou/steps/IOUAmountPage.js#L107
to
```
<Text style={styles.iouAmountText}>{this.props.translations.numberFormat(this.state.amount)}</Text>
```
Here is the list of most places where we can apply these methods.
1. https://github.com/Expensify/Expensify.cash/blob/9fbf82e8237b7d1ff416619970038cc0d5bf9252/src/pages/signin/ChangeExpensifyLoginLink.js#L28
2. https://github.com/Expensify/Expensify.cash/blob/9fbf82e8237b7d1ff416619970038cc0d5bf9252/src/components/IOUBadge.js#L31
3. https://github.com/Expensify/Expensify.cash/blob/a75a4077b526417c09ed07316334a707cd2feea2/src/pages/home/report/ReportActionItemDate.js#L14
4. 
 *Updating Shortly after answers to the Questions*
 
## Questions:
@iwiznia 
1. Do we need to replace the phone number and input values as well. For eg: on the login form, we have an input field where the user can type the phone number to log in?

**Comment by iwiznia** [comment]
> We replace all the places where we see number and time date with the following method calls

Watch out because the amounts examples are not quite correct. When we are dealing with a money amount, we need to format it as so by passing the `currency` and `style:'currency'` to the options of the numberFormat call. You can see some examples [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat).

> Do we need to replace the phone number and input values as well. For eg: on the login form, we have an input field where the user can type the phone number to log in?

Good question, I think that on login we should not do anything, but I think it would be nice if we allowed you to for example start a new chat with a phone number by only typing the number without the international prefix and then calling `fromLocalPhone`

> I see that Phone number is also visible at LHN like this. Do we need to added the methods there as well>

Yes

Proposal looks good, @jboniface can you go ahead and hire @parasharrajat in upwork?

**Comment by parasharrajat** [comment]
Yeah. That's one of the Idea as for some of the locales currency sign is shown after the amount.

**Comment by iwiznia** [comment]
@parasharrajat what is the status of this?

**Comment by parasharrajat** [comment]
@iwiznia Sorry for delay on this. I will add the PR today.

**Comment by parasharrajat** [comment]
@iwiznia Question:
1. We have `IOUBadge` which currently shows the requested amount but it has a fixed $ Sign. What should we do here? should take the currency sign from `numberFormat` method, but for that, we need a currency parameter. 
https://github.com/Expensify/Expensify.cash/blob/bbd386709f6d3a25d4b62f7889a3213b6ee45e12/src/components/IOUBadge.js#L31

**Comment by iwiznia** [comment]
hmmmm the currency should be in the report stored in Onyx, so I guess you can use `props.iouReport.currency`, no?

**Comment by parasharrajat** [comment]
Not sure what is the data structure for that and how that works so it's hard to imagine. If you can lead me to any docs on IOU, I can better understand the amount used on different components under different contexts. 

let me check this. Thanks.

**Comment by iwiznia** [comment]
All reports have the same data structure, which is generated by [getSimplifiedReportObject](https://github.com/Expensify/Expensify.cash/blob/e9f2944888877d83412c6463ff4a3b13ab118960/src/libs/actions/Report.js#L198-L222). The IOUs in particular are set [here](https://github.com/Expensify/Expensify.cash/blob/5a7a25ae499ed3b811c981b15a2d437e0aff8947/src/libs/actions/IOU.js#L50-L57) and [here](https://github.com/Expensify/Expensify.cash/blob/e9f2944888877d83412c6463ff4a3b13ab118960/src/libs/actions/Report.js#L345-L357)

**Comment by parasharrajat** [comment]
@iwiznia Here on LHN. Do I need to remove the country code from the 
![Screenshot 2021-05-21 02:05:13](https://user-images.githubusercontent.com/24370807/119046053-dfe06900-b9d9-11eb-8f9e-52d483bf62b5.png)
phone number. 

There are a couple of other places where we show phone numbers like this.

**Comment by iwiznia** [comment]
> There are a couple of other places where we show phone numbers like this.
> In general, do I need to remove the country code from all the places where the phone number is visible?

Yep

> Num.numberformat is important as we receive the amount of 1000 for $10 from the backend and it will give us a string like $10.00.We can't pass that string to localized numberFormat How should I handle this amount to use in localized NumberFormat.

You need to use the locale aware method. If you are using the withLocalize HOC, then it would be: `this.props.numberFormat(props.iouReport.total / 100, {type: 'currency', currency: props.iouReport.currency})`

**Comment by parasharrajat** [comment]
Questions: 
1. In the below image, the country code of the phone number is of the user whose number is shown there? but if we want to remove the +{countryCode}, our localized `toLocalPhone` will not work as it works only for the user who is logged in. `WithLocalize` usages the locale of the current user. And thus I believe we don't want to remove the country code from these places otherwise logged-in users won't be able to know what is the other user's exact country? Does it make sense?

https://user-images.githubusercontent.com/24370807/119046053-dfe06900-b9d9-11eb-8f9e-52d483bf62b5.png

This makes me wonder that there are very few places we need to convert numbers and amounts. And we need to revise the list once more.

**Comment by iwiznia** [comment]
The idea behind the toLocalPhone is to not see the country code of phones in your location. So if I am in USA I would not see all US phones prefixed with +1 but I would see all the non US phones with their prefix. Or in other words, if a phone does not have a prefix, it's because it's in your own country.

**Comment by parasharrajat** [comment]
Gotcha. Adding now. Then I will add you on the PR to review.
